### PR TITLE
apt_repository add falalback to  deprecated apt-key usage

### LINKED ISSUE #77556: apt_repository has fallen behind apt-add-repository's behavior regarding PPAs as of Ubuntu 22.04
### Summary

With 22.04 right around the corner I've been testing some ansible on it to prepare for deploying 22.04 hosts. I've got everything worked out except for adding PPAs.

`apt_repository` is still adding the PPA keys to `/etc/apt/trusted.gpg` instead of placing individual key files in `/etc/apt/trusted.gpg.d/`.

`apt-add-repository` puts the individual key files in `/etc/apt/trusted.gpg.d`.

This move has been ongoing for many years with the previous behavior being deprecated and generating warnings.

### Issue Type

Bug Report

### Component Name

apt_repository

### Ansible Version

```console
$ ansible --version
ansible 2.10.8
  config file = None
  configured module search path = ['/home/wonko/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]
```

### Configuration

```console
$ ansible-config dump --only-changed
```

### OS / Environment

Ubuntu 22.04 (beta)

### Steps to Reproduce

```yaml (paste below)
    - name: Add PPA
      apt_repository:
        repo: ppa:git-core/ppa
        state: present
```

### Expected Results

PPA is added to `/etc/apt/sources.list.d` and gpg key file to be place in `/etc/apt/trusted.gpg.d`

### Actual Results

```console
W: http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/jammy/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by bcoca**
Fixes #77556

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

apt_repository

### CODE REVIEW

File: lib/ansible/modules/apt_repository.py [hunk 1]
```diff
@@ -461,16 +483,32 @@ def add_source(self, line, comment='', file=None):
...
+            if not self._key_already_exists(info['signing_key_fingerprint'], gpg_bin):
+
+                # TODO: report file that would have been added if not check_mode
+                keyfile = '/usr/share/keyrings/%s-%s-%s.gpg' % (os.path.basename(source).replace(' ', '-'), ppa_owner, ppa_name)
+                if not self.module.check_mode:
+                    command = [gpg_bin, '--no-tty', '--keyserver', 'hkp://keyserver.ubuntu.com:80', '--export', info['signing_key_fingerprint']]
+
+                    rc, stdout, stderr = self.module.run_command(command, check_rc=True, encoding=None)
+                    if len(stdout) == 0:
+                        self.module.fail_json(msg='Unable to get required signging key', rc=rc, stderr=stderr, command=command)
```
Comment (mattclay) [inline lib/ansible/modules/apt_repository.py]:
```suggestion
                        self.module.fail_json(msg='Unable to get required signing key', rc=rc, stderr=stderr, command=command)
```

File: lib/ansible/modules/apt_repository.py [hunk 2]
```diff
@@ -461,16 +483,32 @@ def add_source(self, line, comment='', file=None):
...
+                    try:
+                        with open(keyfile, 'wb') as f:
+                            f.write(stdout)
+                        self.module.log('Added repo key "%s" for apt to file "%s"' % (info['signing_key_fingerprint'], keyfile))
+                    except (OSError, IOError) as e:
+                        self.module.fail_json(msg='Unable to add required singing key for%s ', rc=rc, stderr=stderr, error=to_native(e))
```
Comment (mattclay) [inline lib/ansible/modules/apt_repository.py]:
```suggestion
                        self.module.fail_json(msg='Unable to add required signing key for%s ', rc=rc, stderr=stderr, error=to_native(e))
```

**Comment by mkrizek** [comment]
!waffling core_review

**Pull request merged**